syntax = "proto2";
import "enums.proto";

enum action {
  register_farmer= 0;
  register_buyer= 1;
  register_transporter= 2;
  otp_transaction= 3;
}


message Register_farmer{
  required string aadhar_card = 1;
  required uint64 timestamp = 2;    //will not be updated
  required string full_name = 3;
  required state State = 4;
  required uint32 pincode = 5;
  required uint32 mobilenumber = 6;
  required string district= 7;
  optional bytes photo = 8;      //change to required in production build
  required uint32 otp = 9;
}

message Register_buyer{
  required string aadhar_card = 1;
  required uint64 timestamp = 2;    //will not be updated
  required string full_name = 3;
  required state State = 4;
  required uint32 pincode = 5;
  required uint32 mobilenumber = 6;
  required string district = 7;
  optional bytes photo = 8;   
  required uint32 otp =9;
}

message Register_transporter{
  required string aadhar_card = 1;
  required uint64 timestamp = 2;    //will not be updated
  required string full_name = 3;
  required state State = 4;
  required uint32 pincode = 5;
  required uint32 mobilenumber = 6;
  required string district = 7;
  required string driving_license = 8;
  optional bytes photo = 9;       //change to required in production build
  required uint32 otp = 10;
}

message Otp_transaction{
    required string mobilenumber = 1;
    required uint32 otp =2;
    required uint32 timestamp=3;
}

message Realpayload{
    //ACTION WILL STATE WHAT TO EXPECT IN PAYLOAD
    required action Action = 1;
    // ANYONE OF THE FOLLOWING ACTIONS WILL BE SELECTED
    optional Register_farmer reg_far = 2;
    optional Register_buyer reg_buy = 3;
    optional Register_transporter reg_tra = 4;
    optional Otp_transaction otp_tra = 5;
}
